### ISSUE TITLE
hyperparameter search terminates early in the event of a run fail during a search

### ISSUE THREAD

**Issue by gryBox** [guildai/guildai#21]
Hi @gar1t  - Thank you for a great package. I have encountered two errors that terminate a hyperparameter search, mid-search.  In both cases it was due to wide a search. 
The first error is `skopt` and the second returned another error. In both cases the `objective-loss` resulted in a `nan`.

**Comment by gar1t** [comment]
@gryBox could you provide any additional information about the errors? I'd like to make sure I get the right fix so I'll need to reproduce both issues.

I have your project with default flag values so I should be able to see something breaking - but it'd be helpful to confirm we're seeing the same error.

**Comment by gryBox** [comment]
@gar1t Below is a Traceback from a hyperparameter search terminated early.  It looks like some bad parameter was initialized and that stopped the next search from occurring.  I will continue to dig up a couple more examples.

Separately: `guild publish info x --output --guild-issues`? 

Config
```
Using config:
  {'B_VALIDATION': False,
   'CONFIG_NAME': 'DAMSM',
   'CUDA': True,
   'DATASET_NAME': 'photosynthesis',
   'DATA_DIR': 'photosynthesis',
   'GAN': {'B_ATTENTION': True,
           'B_DCGAN': False,
           'CONDITION_DIM': 100,
           'DF_DIM': 64,
           'GF_DIM': 128,
           'R_NUM': 2,
           'Z_DIM': 100},
   'GPU_ID': 0,
   'OUTPUT_DIR': 'output',
   'RNN_TYPE': 'LSTM',
   'TEXT': {'CAPTIONS_PER_IMAGE': 1, 'EMBEDDING_DIM': 315, 'WORDS_NUM': 166},
   'TRAIN': {'BATCH_SIZE': 7,
             'B_NET_D': True,
             'DISCRIMINATOR_LR': 0.0002,
             'ENCODER_LR': 0.5405147588369471,
             'FLAG': True,
             'GENERATOR_LR': 0.0002,
             'MAX_EPOCH': 201,
             'NET_E': '',
             'NET_G': '',
             'RNN_GRAD_CLIP': 0.3288562144497824,
             'SMOOTH': {'GAMMA1': 266.4302769852824,
  INFO: [root] Number of text files: 7
                        'GAMMA2': 36.21473066134876,
                        'GAMMA3': 65.22665073103775,
  DEBUG: [root] Deleting file: photosynthesis/captions.pickle
                        'LAMBDA': 1.0},
             'SNAPSHOT_INTERVAL': 100},
   'TRAIN_SPLIT': 1.0,
   'TREE': {'BASE_SIZE': 5, 'BRANCH_NUM': 1},
   'VALIDATION_SPLIT': 1.0,
   'WORKERS': 1}
  photosynthesis/filenames.txt
  Number of Train files: 7
  Number of Test files: 7
```
Traceback:
```
DEBUG: [root] Loading pickle text file names: photosynthesis/captions.pickle
  /home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/torch/nn/modules/rnn.py:46: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.5 and num_layers=1
    "num_layers={}".format(dropout, num_layers))
  Load pretrained model from  https://download.pytorch.org/models/inception_v3_google-1a9a5a14.pth
  Traceback (most recent call last):
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/runpy.py", line 193, in _run_module_as_main
      "__main__", mod_spec)
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/runpy.py", line 85, in _run_code
      exec(code, run_globals)
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/op_main.py", line 272, in <module>
      main()
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/op_main.py", line 53, in main
      _main()
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/op_main.py", line 80, in _main
      _try_module(arg1, rest_args)
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/op_main.py", line 129, in _try_module
      _dispatch_module_exec(_flags_dest(args), module_info)
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/op_main.py", line 202, in _dispatch_module_exec
      _exec_module_with_args(module_info, args)
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/op_main.py", line 210, in _exec_module_with_args
      _module_main(module_info)
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/op_main.py", line 221, in _module_main
      _gen_exec(module_info, main)
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/op_main.py", line 234, in _gen_exec
      exec_cb()
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/site-packages/guild/op_main.py", line 220, in main
      imp.load_module("__main__", f, path, desc)
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/imp.py", line 235, in load_module
      return load_source(name, filename, file)
    File "/home/ec2-user/anaconda3/envs/py36-ml/lib/python3.6/imp.py", line 170, in load_source
      module = _exec(spec, sys.modules[name])
    File "<frozen importlib._bootstrap>", line 618, in _exec
    File "<frozen importlib._bootstrap_external>", line 678, in exec_module
    File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
    File "/home/ec2-user/environment/AttnGAN/code/pretrain_DAMSM.py", line 382, in <module>
      dataset.ixtoword, image_dir)
    File "/home/ec2-user/environment/AttnGAN/code/pretrain_DAMSM.py", line 109, in train
      cap_lens, class_ids, batch_size)
    File "/home/ec2-user/environment/AttnGAN/code/miscc/losses.py", line 91, in words_loss
      weiContext, attn = func_attention(word, context, cfg.TRAIN.SMOOTH.GAMMA1)
    File "/home/ec2-user/environment/AttnGAN/code/GlobalAttention.py", line 48, in func_attention
      attn = torch.bmm(contextT, query) # Eq. (7) in AttnGAN paper
  RuntimeError: invalid argument 6: wrong matrix size at /opt/conda/conda-bld/pytorch_1549628766161/work/aten/src/THC/generic/THCTensorMathBlas.cu:492

```

**Comment by gar1t** [comment]
Let's see how this operation fares now with `0.6.4.rc3`. I tried running the check on AttnGAN but I'm running into some issues with the newly added dataset support [1].

[1] https://github.com/gryBox/AttnGAN/blob/photo/code/published-runs/cabadc088e8a11e99e4cc85b764bbf34/README.md#output

**Comment by gar1t** [comment]
Closing this - if we see any more of this we can reopen.
